cmd/spire-server/cli/bundle/delete.go

```diff
@@ -7,47 +7,40 @@ import (
 	"fmt"
 
 	"github.com/mitchellh/cli"
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
 	"github.com/spiffe/spire/pkg/common/idutil"
-	"github.com/spiffe/spire/proto/spire/api/registration"
-)
-
-const (
-	deleteBundleRestrict   = "restrict"
-	deleteBundleDissociate = "dissociate"
-	deleteBundleDelete     = "delete"
+	"github.com/spiffe/spire/proto/spire/api/server/bundle/v1"
+	"google.golang.org/grpc/codes"
 )
 
 // NewDeleteCommand creates a new "delete" subcommand for "bundle" command.
 func NewDeleteCommand() cli.Command {
-	return newDeleteCommand(defaultEnv, newClients)
+	return newDeleteCommand(common_cli.DefaultEnv)
 }
 
-func newDeleteCommand(env *env, clientsMaker clientsMaker) cli.Command {
-	return adaptCommand(env, clientsMaker, new(deleteCommand))
+func newDeleteCommand(env *common_cli.Env) cli.Command {
+	return util.AdaptCommand(env, new(deleteCommand))
 }
 
 type deleteCommand struct {
 	// SPIFFE ID of the trust domain bundle
 	id string
-
-	// Deletion mode.
-	mode string
```

Comment:
New api no longer provide options when removing a bundle, it use RESTRICT mode as default, 
based on that I removed mode from `spire-server bundle delete`

Reply:
I don't think that we can just remove the `-mode` flag without going through the usual deprecation cycle. I see some possible alternatives:
1. if we want to get rid of the old API in 0.12.0, we may try to keep the `-mode` flag in the `spire-server bundle delete` command with a deprecation notice and somehow implement the functionality for DISSOCIATE and DELETE.
2. port all the commands except for `spire-server bundle delete` where we add the deprecation notice for the `-mode` flag and finish the migration in a later release.

I'm gathering opinions from other maintainers on this topic.

Reply:
We discussed this in today's maintainer's call. We decided that we need to restore the ability to specify the mode in the `BatchDeleteFederatedBundle` RPC. I filed #1895 for that and @MarcosDY offered to take care of it. Thanks @MarcosDY!